### PR TITLE
Bundles size optimizations

### PR DESCRIPTION
**Pull request by blurbyte**
## Proposed changes

Removed **lodash** wherever it made sense or where it was not used at all. Whole bundle size went down by **~600KB** uncompressed.
Replaced **tinycolor2** with custom function. **tinycolor2** is only left in **Text** package.
Passed **ref** to Switch component.

TODO:
Replace tinycolor2 in Text.

Before:
lodash-before

After:
lodash-after

## Functional Code

- [x] Builds without errors
- [x] Linter passes CI
- [x] Unit tests written & pass CI
- [x] Integration tests written & pass CI
- [x] Tested on [supported browsers](https://support.datacamp.com/hc/en-us/articles/360001541574-Minimum-System-Requirements)

## Documentation

- [ ] ~~Technical docs written~~
- [ ] ~~Structural changes reflected in Readme~~
- [ ] ~~Migration plan for breaking changes~~

## Meets Product Requirement

- [x] Assumptions are met
- [x] Meets acceptance criteria
- [ ] ~~Approved by Designer, Engineer, & PO~~

### CODE REVIEW

**Review by dandenney — APPROVED** [review]
This is rad, Lodash is as bloated as it is helpful

**Review by hermansje — APPROVED** [review]
🎉

File: packages/react-components/button/src/Button/buttonStyles.ts [hunk 1]
```diff
@@ -30,8 +29,15 @@ const primaryHoverColors: IntentMap = {
...
 };
 
-const outlineHoverColors: IntentMap = _.mapValues(baseColors, (value) =>
-  tinycolor(value).setAlpha(0.15).toRgbString(),
+const outlineHoverColors: IntentMap = Object.entries(baseColors).reduce(
+  (hoverColors, baseColor) => {
+    const [intent, value] = baseColor;
+    return {
+      ...hoverColors,
+      [intent]: tinycolor(value).setAlpha(0.15).toRgbString(),
```
[the remark this answers is not in the document]
Comment (blurbyte) [reply packages/react-components/button/src/Button/buttonStyles.ts:37]: I 100%  agree, the future tokens are not that intuitive to use, and honestly I prefer the simplicity of the old ones.

**Comment by blurbyte** [comment]
@dandenney Yeah. The first thing I tried was importing individual lodash functions from separate npm modules, but it didn't help that much unfortunately. The conclusion is: do not use lodash in any shape or form :)

**Review by hermansje — APPROVED** [review]
LGTM!

**Pull request merged**